src/Cargo.lock

```diff
@@ -1723,6 +1723,72 @@ dependencies = [
  "syntax_pos 0.0.0",
 ]
 
+[[package]]
+name = "rustc-ap-rustc_cratesio_shim"
+version = "12.0.0"
```

Comment:
Since we are building rustfmt in the compiler, it's strange we in turn depend on an external, older copy of these crates; besides, these will almost always be outdated (the current version is 14, for instance).

The advantage of this is that compiler changes will less likely to break rustfmt on nightly. 

Not sure if we want to r+ as-is or require a `[patch]` section.

Reply:
I also wonder how this affects [build times](https://github.com/rust-lang/rust/issues/46903#issuecomment-357061260).

Reply:
@petrochenkov It shouldn't affect build times since those two builds doesn't touch rustfmt/rls.

Reply:
@kennytm same thing [will happen](https://github.com/rust-lang-nursery/rls/pull/660/files#diff-d2ede298d9a75c849170d6ef285eda1eR995) to rls, however rls uses a [locally-patched](https://github.com/rust-lang/rust/blob/0692b3df2e3f9f5e0d3ba7505dbdd667e9822f9e/src/Cargo.toml#L62) version of rustfmt, so I assume it'll work as intended, provided that rustfmt compiles as expected in-tree, with `[patch]`ed deps or not, no?

Reply:
It is intentional to depend on an older version of the crate, it means that breakage in libsyntax doesn't break rustfmt (and the RLS), which is the whole reason for doing this. Using `[patch]` would prevent that happening.